src/luks2_token.rs

```diff
@@ -48,24 +48,42 @@ impl<'a> CryptLuks2Token<'a> {
     pub fn json_set(
         &mut self,
         token: Option<c_uint>,
-        json: &serde_json::Value,
+        json: Option<&serde_json::Value>,
     ) -> Result<c_uint, LibcryptErr> {
-        let json_cstring =
-            to_cstring!(serde_json::to_string(json).map_err(LibcryptErr::JsonError)?)?;
-        errno_int_success!(unsafe {
-            libcryptsetup_rs_sys::crypt_token_json_set(
-                self.reference.as_ptr(),
-                token
-                    .map(|t| t as c_int)
-                    .unwrap_or(libcryptsetup_rs_sys::CRYPT_ANY_TOKEN),
-                json_cstring.as_ptr(),
-            )
-        })
+        if let Some(json) = json {
+            let json_cstring =
+                to_cstring!(serde_json::to_string(json).map_err(LibcryptErr::JsonError)?)?;
+            errno_int_success!(unsafe {
+                libcryptsetup_rs_sys::crypt_token_json_set(
+                    self.reference.as_ptr(),
+                    token
+                        .map(|t| t as c_int)
+                        .unwrap_or(libcryptsetup_rs_sys::CRYPT_ANY_TOKEN),
+                    json_cstring.as_ptr(),
+                )
+            })
+        } else {
+            let token_slot = token.map(|t| t as c_int).ok_or_else(|| {
+                LibcryptErr::Other(
+                    "json_set: both token and json can't be None simultaneously".into(),
+                )
+            })?;
+            errno_int_success!(unsafe {
+                libcryptsetup_rs_sys::crypt_token_json_set(
+                    self.reference.as_ptr(),
+                    token_slot,
+                    ptr::null(),
+                )
+            })
+        }
         .map(|rc| rc as c_uint)
     }
 
     /// Get the token info for a specific token
-    pub fn status(&mut self, token: c_uint) -> Result<(CryptTokenInfo, String), LibcryptErr> {
+    pub fn status(
+        &mut self,
+        token: c_uint,
+    ) -> Result<(CryptTokenInfo, Option<String>), LibcryptErr> {
```

Comment:
It may make sense to simply change `CryptTokenInfo` to have newtype variants containing a `String` for all of the variants other than `Invalid` and `Inactive` as those won't return a type string.

Reply:
That's indeed an cleaner solution, I'm however not fluent enough in macro magic to adapt the `consts_to_from_enum` macro to handle this usecase.

Reply:
That's not a problem. This use case is outside of what that macro is intended for.